hotsos/core/plugins/kernel/kernlog/common.py

```diff
@@ -1,15 +1,47 @@
 import abc
 import os
 
+from searchkit.constraints import TimestampMatcherBase
+
+from hotsos.core.log import log
 from hotsos.core.config import HotSOSConfig
 from hotsos.core.host_helpers import CLIHelper, HostNetworkingHelper
-from hotsos.core.search import FileSearcher
+from hotsos.core.search import FileSearcher, SearchConstraintSearchSince
 
 KERNLOG_TS = r'\[\s*\d+\.\d+\]'
 KERNLOG_PREFIX = (r'(?:\S+\s+\d+\s+[\d:]+\s+\S+\s+\S+:\s+)?{}'.
                   format(KERNLOG_TS))
 
 
+class KernLogTimestampMatcher(TimestampMatcherBase):
+    """
+    kern.log has a slightly esoteric timetamp format so we have to do a little
+    juggling to get it into a standard format that can be used with search
+    constraints.
+    """
+
+    @property
+    def year(self):
+        """ Use current year as it is not normally included in the logs """
+        return CLIHelper().date(format='+%Y')
+
+    @property
+    def month(self):
+        _month = self.result.group('month').lower()
+        for sm, im in {'jan': 1, 'feb': 2, 'mar': 3, 'apr': 4, 'may': 5,
+                       'jun': 6, 'jul': 7, 'aug': 8, 'sep': 9, 'oct': 10,
+                       'nov': 11, 'dec': 12}.items():
+            if _month.startswith(sm):
+                return im
```

Comment:
Just a question on this code. I know there are many ways to do this, and this works fine, but I was wondering of this was called a lot, and are we going to be iterating through this list a lot?  If so it could be tweaked to avoid that, something like:

months = {'jan': 1, ... } (define at top of file)
_month = self.result.group('month').lower()[:3]
try:
    return months[_month]
except KeyError:
    log.error(...)

Reply:
in reality it will only be executed once per instance but i have tidied up a bit and moved to be a class attribute